code/modules/overmap/ships/engines/thermal.dm

```diff
@@ -99,12 +99,20 @@
 	if(!removed)
 		return 0
 	. = calculate_thrust(removed)
-	playsound(loc, 'sound/machines/thruster.ogg', 100 * thrust_limit, 0, world.view * 4, 0.1)
+	playsound(get_turf(src), 'sound/machines/thruster.ogg', 100 * thrust_limit, 0, world.view * 4, 0.1)
 	var/exhaust_dir = reverse_direction(dir)
 	var/turf/T = get_step(src,exhaust_dir)
+	var/range = clamp(round(.), 1, 8)
 	if(T)
 		T.assume_air(removed)
-		new/obj/effect/engine_exhaust(T, exhaust_dir, air_contents.check_combustability() && air_contents.temperature >= PLASMA_MINIMUM_BURN_TEMPERATURE)
+		// if its 1000k hot
+		new/obj/effect/engine_exhaust(T, exhaust_dir, air_contents.temperature >= 1000, (range > 1))
+		for(var/i=1,i < range,i++)
+			T = get_step(T,exhaust_dir)
+			if(!T)
+				break
+			sleep(0.5)
+			new/obj/effect/engine_exhaust(T, exhaust_dir, air_contents.temperature >= 1000, (i != range - 1))
```

Comment:
It would be better if this was a callback timer. Also, based on some reading, sleep() and spawn() will truncate decimal values.
```suggestion
			addtimer(CALLBACK(src, .proc/extend_plume, T, exhaust_dir, air_contents.temperature >= 1000, i != range - 1), i - 1, TIMER_UNIQUE)

/obj/machinery/atmospherics/unary/engine/proc/extend_plume(turf/T, exhaust_dir, is_flame, is_midsection)
	new/obj/effect/engine_exhaust(T, exhaust_dir, is_flame, is_midsection)
```